app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_with_person_inst = [
+        Person(name=person["name"], age=person["age"])
+        for person in people
+    ]
+    for human in people:
+        person = Person.people[human["name"]]
+        if "wife" in human and human["wife"] is not None:
+            person.wife = Person.people[human["wife"]]
+        elif "husband" in human and human["husband"] is not None:
+            person.husband = Person.people[human["husband"]]
```

Comment:
Similar to the comment on line 18, dynamically adding the `husband` attribute can lead to attribute errors. It's better to initialize this attribute in the `Person` class constructor with a default value.